tests/models/test_lora.py

```diff
@@ -0,0 +1,89 @@
+from typing import List
+
+import numpy as np
+import pytest
+import tensorflow as tf
+
+from tfimm.architectures import ConvNeXt, ConvNeXtConfig, lora
+from tfimm.models import create_model, register_model
+
+
+# We test both rank 2 and rank 3 inputs, since they use different paths in call().
+@pytest.mark.parametrize("input_shape", [(3, 4), (3, 4, 5)])
+def test_lora_dense(input_shape):
+    """Creating the layer and running inference in merged and non-merged modes."""
+    layer = lora.LoRADense(units=2, use_bias=True, activation="swish")
+    layer.build(input_shape=input_shape)
+    assert not layer.merged
+
+    # Set non-trivial weights, since bias and kernel_b are zero-initialised.
+    layer.bias = tf.random.uniform(layer.bias.shape)
+    layer.kernel_lora_a = tf.random.uniform(layer.kernel_lora_a.shape)
+    layer.kernel_lora_b = tf.random.uniform(layer.kernel_lora_b.shape)
+
+    x = tf.random.uniform(input_shape)
+    res_1 = layer(x)
+
+    layer.merge_weights()
```

Comment:
can you add a check before the call to `merge_weights` that `assert not layer.merged`?

Reply:
Sure.